### ISSUE TITLE
More Than One Hidden Layer Does Not Work

### ISSUE THREAD

**Issue by devonlee111** [devonlee111/Basic-Neural-Net#1]
**<h1>Problem</h1>**
The NN.py program does not train properly when using 2 or more hidden layers.

**<h1>Possible Errors</h1>**
*The cause of this problem is currently unknown.*
1. Most likely a problem with the backpropagation algorithm.
   1. Backpropagation function may not be implemented fully/completely correct.
      1. Error is most likely a problem where the layer error is incorrectly calculated.
   1. There may be a minor error in the code for the backpropagation function.
1. Potentially a slight code/logic error in forward pass method.

**<h1>Methods To Find A Solution</h1>**
*More research will be done on the backpropagation function to see what may have been implemented incorrectly. 
*A review of the code will be performed in case it turns out to be a minor algorithm error.

**Comment by devonlee111** [comment]
<h1><strong>Solution</strong></h1>
It seems this problem wasn't really an implementation problem at all. It seems that the real issue was using the sigmoid function as an activation function. When using sigmoid on a neural net with more than a single hidden layer, the neural net exhibited a severe case of, what I believe to be, vanishing gradients. I have solved this problem by implementing the option of using tanh as the activation function. Using tanh seems to have removed the vanishing gradients problem when using multiple hidden layers.
